File: sephora-backend/Application/Dtos/Product/ProductDTO.cs
namespace CleanArchitecture.Application.Dtos.Product;

public class ProductDto
{
    public long Id { get; set; }

    public string Name { get; set; } = String.Empty;

    public string DescriptionEn { get; set; } = String.Empty;

    public string DescriptionUa { get; set; } = String.Empty;

    public bool Active { get; set; }

    public BrandDto Brand { get; set; } = default!;

    public CategoryDto Category { get; set; } = default!;

    public IEnumerable<ProductPieceDto> Pieces { get; set; } = [];

    public IEnumerable<RatingDto> Ratings { get; set; } = [];

    public decimal AverageRating { get; set; }

    public IEnumerable<AmountDto> Volumes { get; set; } = [];

    public IEnumerable<CharacteristicDto> Characteristics { get; set; } = [];

    public DateTime CreatedAt { get; set; }

    // Publication was less than 14 days ago
    public bool IsNew => CreatedAt > DateTime.UtcNow.AddDays(-14);

    public bool IsFavorite { get; set; }
}

File: sephora-frontend/src/models/product/ProductDto.ts
import BrandDto from "../brand/BrandDto.ts";
import RatingDto from "../rating/RatingDto.ts";
import ProductPieceDto from "../piece/ProductPieceDto.ts";
import AmountDto from "../amount/AmountDto.ts";
import CategoryDto from "../category/CategoryDto.ts";
import CharacteristicDto from "../characteristics/CharacteristicDto.ts";

export default interface ProductDto {
    get id(): number;

    get name(): string | null;

    get descriptionEn(): string;

    get descriptionUa(): string;

    get active(): boolean;

    get brand(): BrandDto;

    get category(): CategoryDto;

    get pieces(): ProductPieceDto[];

    get ratings(): RatingDto[];

    get averageRating(): number;

    get volumes(): AmountDto[];

    get characteristics(): CharacteristicDto[];

    get createdAt(): Date;

    get isNew(): boolean;

    get isFavorite(): boolean;
}
